orte/mca/errmgr/default_hnp/errmgr_default_hnp.c

```diff
@@ -85,6 +93,129 @@ static void default_hnp_abort(orte_job_t *jdata);
 static void job_errors(int fd, short args, void *cbdata);
 static void proc_errors(int fd, short args, void *cbdata);
 
+static int pack_state_for_proc(opal_buffer_t *alert, orte_proc_t *child)
+{
+    int rc;
+
+    /* pack the child's vpid */
+    if (ORTE_SUCCESS != (rc = opal_dss.pack(alert, &(child->name.vpid), 1, ORTE_VPID))) {
+        ORTE_ERROR_LOG(rc);
+        return rc;
+    }
+    /* pack the pid */
+    if (ORTE_SUCCESS != (rc = opal_dss.pack(alert, &child->pid, 1, OPAL_PID))) {
+        ORTE_ERROR_LOG(rc);
+        return rc;
+    }
+    /* pack its state */
+    if (ORTE_SUCCESS != (rc = opal_dss.pack(alert, &child->state, 1, ORTE_PROC_STATE))) {
+        ORTE_ERROR_LOG(rc);
+        return rc;
+                                            }
+    /* pack its exit code */
+    if (ORTE_SUCCESS != (rc = opal_dss.pack(alert, &child->exit_code, 1, ORTE_EXIT_CODE))) {
+        ORTE_ERROR_LOG(rc);
+        return rc;
+    }
+
+    return ORTE_SUCCESS;
+}
+
+static void register_cbfunc(int status, size_t errhndler, void *cbdata)
```

Comment:
What is the need for an empty callback ?

Reply:
With this i know the event registration is succeed. If you think this is not good, I can use NULL when i register.